src/sparseml/transformers/finetune/data/custom.py

```diff
@@ -12,13 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from copy import deepcopy
-from typing import List
+from typing import Callable, List
 
 from datasets.dataset_dict import DatasetDict
 
+from sparseml.base import import_attr_from_path
 from sparseml.transformers.finetune.data import TextGenerationDataset
 
 
+PREPROCESSING_FUNCTION_REGISTRY_PATH = (
+    "sparseml.transformers.utils.preprocesing_funtions"
+)
+
+
```

Comment:
Instead of storing the registry as a constant variable, why not implement it as a RegistryMixIn? Then we can tag each helper function to the registry and avoid all that string parsing of the path

Reply:
Ok sounds good. RegistryMixIn for prepreprocessing func is created!